addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CrossSiteScriptingScanRule.java

```diff
@@ -240,6 +242,23 @@ private List<HtmlContext> performAttack(
                 findDecoded ? getURLDecode(evidence) : evidence, null, 0, ignoreSafeParents);
     }
 
+    private void setAlert(
+            int confidence,
+            String param,
+            String attack,
+            String evidence,
```

Comment:
Pass `HtmlContext` instead.

Reply:
there are cases where ctx.get(0).getTarget() has been used. But its focused with this PR so :)

Reply:
I'm aware but the ones that are mostly different can stay the same, using the `HtmlContext` reduces the duplication for more cases than having the multiple args.